tests/Core/Lock/LockTest.php

```diff
@@ -0,0 +1,116 @@
+<?php declare(strict_types=1);
+/**
+ * This File is part of JTL-Software
+ *
+ * User: avermeulen
+ * Date: 2021-01-26
+ */
+
+namespace Core\Lock;
+
+use JTL\SCX\Lib\Channel\Contract\Core\Lock\LockProvider;
+use JTL\SCX\Lib\Channel\Core\Lock\Lock;
+use MongoDB\Exception\RuntimeException;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @covers \JTL\SCX\Lib\Channel\Core\Lock\Lock
+ */
+class LockTest extends TestCase
+{
+    public function testCanObtain(): void
+    {
+        $now = new \DateTimeImmutable();
+        $lockKey = uniqid('lockKey', true);
+        $ttl = random_int(1, 99);
+        $lockProvider = $this->createMock(LockProvider::class);
+        $lockProvider->expects(self::once())->method('obtain')
+            ->with($lockKey, self::callback(function (\DateTimeImmutable $expiresAt) use ($now, $ttl) {
+                return $now->getTimestamp() + $ttl === $expiresAt->getTimestamp();
```

Comment:
Das kommt in jeden Test vor und ist in meinen Augen absolut unnötig. bzw. sollte einen eigenen Test erhalten
Vorschlag: `it_will_calculate_ttl_correctly`

Die anderen Tests sollten sich auf das Verhalten der Klasse konzentieren.

* it will return true when a lock could obtain
* it will return false when a lock could not obtain
* ...

Ein obtain()=false bedeutet ja nicht das etwas "fail" ist. Sondern das ein Lock eventuell bereits existiert.

Reply:
Weiterhin `return $now->getTimestamp() + $ttl === $expiresAt->getTimestamp();` wird ggf. auch false liefern

`$this->assertEqualsWithDelta($now->getTimestamp() + $ttl, $expiresAt->getTimestamp(), 1.0)`